blocklylib-core/src/main/java/com/google/blockly/android/ui/fieldview/BasicFieldImageView.java

```diff
@@ -177,14 +193,18 @@ InputStream getStreamForSource(String source) throws IOException {
         }
     }
 
+    @UiThread
     protected void updateViewSize() {
         // Check for null b/c of https://groups.google.com/d/msg/blockly/lC91XADUiI4/Y0cLRAYQBQAJ
-        if (mImageField != null ) {
-            float density = getContext().getResources().getDisplayMetrics().density;
-            int pxWidth = (int) Math.ceil(mImageField.getWidth() * density);
-            int pxHeight = (int) Math.ceil(mImageField.getHeight() * density);
-            setMinimumWidth(pxWidth);
-            setMinimumHeight(pxHeight);
+        // Synchronize ImageField update. Issues #678.
+        synchronized (mImageFieldLock) {
+            if (mImageField != null) {
```

Comment:
If it is null should you set the size to a min default?

Reply:
Set to zero in this case.